code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,309 @@
+//Machine to hold a deployed gun. It aquired nearly all of its variables from the gun itself.
+/obj/machinery/mounted
+
+	name = "Generic Mounted Gun"
+	desc = "big gun"
+	icon = 'icons/obj/items/gun.dmi'
+	icon_state = "gun"
+
+	anchored = TRUE
+	resistance_flags = XENO_DAMAGEABLE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	use_power = FALSE
+	max_integrity = 100
+	soft_armor = list("melee" = 0, "bullet" = 50, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 0, "fire" = 0, "acid" = 0)
+	hud_possible = list(MACHINE_HEALTH_HUD, SENTRY_AMMO_HUD)
+
+	///Gun that does almost all the work when deployed. This type should not exist with a null 'gun'
+	var/obj/item/weapon/gun/gun
+
+	///Flags inhereted from the deployed gun to determin if it can be picked up, or rotated.
+	var/deploy_flags
+
+	///Icon state for when the gun has ammo.
+	var/icon_full = "turret" 
+	///Icon state for when the gun has no ammo.
+	var/icon_empty = "turret_e"
+	//this is amount of tiles we shift our vision towards guns direction when operated, currently its max is 7
+	var/view_tile_offset = 3
+
+/obj/machinery/mounted/Initialize()
+	. = ..()
+	prepare_huds()
+	for(var/datum/atom_hud/squad/sentry_status_hud in GLOB.huds) //Add to the squad HUD
+		sentry_status_hud.add_to_hud(src)
+
+///generates the icon based on how much ammo it has.
+/obj/machinery/mounted/update_icon() 
+	if(icon_empty)
+		if(!gun.current_mag || istype(gun, /obj/item/weapon/gun/launcher))
+			icon_state = "[icon_empty]"
+	if(icon_full)
+		icon_state = "[icon_full]"
+	else
+		icon_state = initial(icon_state)
+	hud_set_machine_health()
+	hud_set_gun_ammo()
+
+///Handles variable transfer from the gun, to the machine.
+/obj/machinery/mounted/proc/deploy(obj/item/weapon/gun/new_gun, direction)
+	if(istype(new_gun.current_mag, /obj/item/ammo_magazine/internal) || istype(new_gun, /obj/item/weapon/gun/launcher))
+		stack_trace("[new_gun] has been deployed, however it is incompatible because of either an internal magazine, or it is a launcher.")
+		return
+	gun = new_gun
+	gun.forceMove(src)
+
+	deploy_flags = gun.deploy_flags
+	obj_integrity = gun.deploy_integrity
+	max_integrity = gun.deploy_max_integrity
+	view_tile_offset = gun.deploy_view_offset
+	
+	if(!gun.deploy_name)
+		name = gun.name
+	else
+		name = gun.deploy_name
+	
+	if(!gun.deploy_desc)
+		desc = gun.desc
+	else
+		desc = gun.deploy_desc
+
+	if(!gun.deploy_icon)
+		icon = gun.icon
+	else
+		icon = gun.deploy_icon
+
+	if(!gun.deploy_icon_state)
+		icon_state = gun.icon_state
+	else
+		icon_state = gun.deploy_icon_state
+
+	if(!gun.deploy_icon_full)
+		icon_full = gun.icon_state
+	else
+		icon_full = gun.deploy_icon_full
+
+	if(!gun.deploy_icon_empty)
+		icon_full = gun.icon_state
+	else
+		icon_full = gun.deploy_icon_empty
```

Comment:
all of this need to be inline with the ? operator.

Exemple :   icon_empty =  gun.deploy_icon_empty ?  gun.icon_state : gun.deploy_icon_empty

And you did a typo, you have two times icon_full